exercises/practice/kindergarten-garden/test_kindergarten_garden.c

```diff
@@ -0,0 +1,164 @@
+#include "kindergarten_garden.h"
+
+#include "test-framework/unity.h"
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static void test_kindergarten_garden(const char *diagram, const char *student,
+                                     const plant_t *expected)
+{
+   plant_t actual[4];
+   plants(diagram, student, actual);
```

Comment:
What if you just returned struct containing an array of size 4? Since it's a known size it's strange to require it to be allocated by the client and then passed as an output parameter.

Reply:
Makes sense, although I'm not entirely happy with the naming. I wanted to return a fixed size array at first, but that doesn't work. Wrapping it in a struct is a good workaround!